bouncer/src/repo/core/model/bson/repo_bson_factory.cpp

```diff
@@ -937,6 +934,21 @@ TextureNode RepoBSONFactory::makeTextureNode(
 		repoWarning << " Creating a texture node with no texture!";
 	}
 
+	//--------------------------------------------------------------------------
```

Comment:
appendDefaults will sort this out for you if you pass the parentIDs in, you don't have to create your own function to put it into the document:
![image](https://user-images.githubusercontent.com/11945337/101914644-72bfe080-3bbc-11eb-97b5-5c12b4f5fb0c.png)

Reply:
done.

Just a side question though:
is it worth removing dupes and ordering before passing it in? This is what the original call to cloneAndAddParent was doing:
https://github.com/3drepo/3drepobouncer/blob/37f7ac3ad8cf13bd403e323ce956378573d87c35/bouncer/src/repo/core/model/bson/repo_node.cpp#L87

Reply:
probably not. cloneAndAddParent is doing this because it has to merge the new parents with the existing parents, therefore it need to ensure there's no duplicates.

In this case in repoBsonFactory we're instantiating a new instance from scratch so we don't really have this issue